### PR TITLE
Mempool mutex

### PR DESCRIPTION
**Pull request by 0xSulpiride**
## Description

- added log on userop deletion
- added mutex on mempool operations

## Types of changes
What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Further comments (optional)

-

### CODE REVIEW

File: packages/executor/src/modules/eth.ts [hunk 1]
```diff
@@ -480,7 +480,7 @@ export class Eth {
         let fromBlockNumber = blockNumber - this.config.receiptLookupRange;
         // underflow check
         if (fromBlockNumber < 0) {
-          fromBlockNumber = blockNumber;
+          fromBlockNumber = 0;
```
Comment (ch4r10t33r) [inline packages/executor/src/modules/eth.ts:483]: Wont this cause delays/timeout?

File: packages/executor/src/services/MempoolService.ts [hunk 2]
```diff
@@ -104,34 +108,41 @@ export class MempoolService {
     if (!entry) {
       return;
     }
-    const key = this.getKey(entry);
-    const newKeys = (await this.fetchKeys()).filter((k) => k !== key);
-    await this.db.del(key);
-    await this.db.put(this.USEROP_COLLECTION_KEY, newKeys);
+    await this.mutex.runExclusive(async () => {
+      this.logger.debug(`${entry?.userOpHash} deleted from mempool`);
```
Comment (ch4r10t33r) [inline packages/executor/src/services/MempoolService.ts:112]: move this after delete.

**Review by ch4r10t33r — APPROVED** [review]
LGTM, pls change and merge.

File: packages/executor/src/modules/eth.ts (hunk 1 above)
Comment (0xSulpiride) [reply packages/executor/src/modules/eth.ts:483]: no, this is for the cases when the blockchain is new and `receiptLookupRange` is more than the number of blocks in blockchain (happens in testing mode only usually)

**Pull request merged**